component/src/main/java/me/androidbox/component/general/TaskActionButton.kt

```diff
@@ -0,0 +1,52 @@
+package me.androidbox.component.general
+
+import androidx.annotation.DrawableRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.BusbyTaskyTheme
+
+@Composable
+fun TaskActionButton(
+    @DrawableRes iconResource: Int,
+    onActionClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
```

Comment:
Whenever I have 1 lambda on a composable, I usually prefer it to be as the last parameter

So:

```
fun TaskActionButton(
    @DrawableRes iconResource: Int,
    modifier: Modifier = Modifier,
    onActionClicked: () -> Unit,
)
```

This is the only case where the optional parameter rule breaks (The rule where the optional parameters need to be after the obligatory parameters)

But only breaks with lambda. Why? Because kotlin has high-order functions

Basically, if you pass a lambda as last parameter, you can then do:

```
TaskActionButton(R.drawable.my_icon) {
// onActionClicked gets called here
} 

```

Reply:
@philipplackner Will change to use named parameters